### PR TITLE
Shift towards context usage in the raw validators

### PR DESCRIPTION
**Pull request by collijk**
I should have done this to start.

### CODE REVIEW

File: src/vivarium_inputs/validation/raw.py [hunk 1]
```diff
@@ -1316,6 +1279,33 @@ def check_paf_types(entity: Union[Etiology, RiskFactor]) -> None:
                       f'{entity.kind} {entity.name} may be outside expected range [0, 1].')
 
 
+def check_cause_age_restrictions_sets(entity: Cause) -> None:
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: why did you move this out of the utilities section?

File: src/vivarium_inputs/validation/raw.py [hunk 2]
```diff
@@ -644,23 +640,23 @@ def validate_deaths(data: pd.DataFrame, entity: Cause, location_id: int, populat
...
+    check_sex_ids(data, context, male_expected, female_expected)
 
-    check_age_restrictions(data, age_group_ids, restrictions.yll_age_group_id_start, restrictions.yll_age_group_id_end)
-    check_sex_restrictions(data, restrictions.male_only, restrictions.female_only)
+    check_age_restrictions(data, context, restrictions.yll_age_group_id_start, restrictions.yll_age_group_id_end)
+    check_sex_restrictions(data, context, restrictions.male_only, restrictions.female_only)
 
     check_value_columns_boundary(data, 0, 'lower', value_columns=DRAW_COLUMNS, inclusive=True, error=DataAbnormalError)
     idx_cols = ['age_group_id', 'year_id', 'sex_id']
+    population = context['population']
```
Comment (kfwilson) [inline src/vivarium_inputs/validation/raw.py]: for filtering pop on line 662 below, you use the sexes in the validation context everywhere else but here is the globals sexes

File: src/vivarium_inputs/validation/raw.py (hunk 1 above)
Comment (collijk) [reply src/vivarium_inputs/validation/raw.py]: Moved from validation utilities to check metadata utilities.
